From: =?UTF-8?q?Rog=C3=A9rio=20Brito?= <rbrito@ime.usp.br>
Date: Wed, 6 Jun 2012 06:10:20 -0300
Subject: Remove FAAC dependency.

We actually kill FAAC completely from here, while we should be ifdef'ing
things (perhaps) and integrating that in upstream's configure system.

With this, we depend on one fewer package of dubious license for handbrake
to enter Debian.
---
 gtk/src/audiohandler.c |    5 +-
 gtk/src/ghb.ui         |    8 --
 gtk/src/hb-backend.c   |    6 +-
 gtk/src/preset_xlat.c  |    2 +-
 gtk/src/presets.c      |    6 +-
 libhb/common.c         |   20 +--
 libhb/common.h         |    1 -
 libhb/encfaac.c        |  318 ------------------------------------------------
 libhb/hb.c             |    2 -
 libhb/internal.h       |    1 -
 libhb/module.defs      |    4 +-
 libhb/muxmkv.c         |    1 -
 libhb/work.c           |    1 -
 13 files changed, 13 insertions(+), 362 deletions(-)
 delete mode 100644 libhb/encfaac.c

--- a/gtk/src/Makefile.am
+++ b/gtk/src/Makefile.am
@@ -3,13 +3,13 @@
 if MINGW
 HB_LIBS= \
 	-lhb -la52 -lmkv -lavformat -lavcodec -lavutil -ldca -ldvdnav -ldvdread \
-	-lfaac -lmp3lame -lmpeg2 -lvorbis -lvorbisenc -logg -lsamplerate \
+	-lmp3lame -lmpeg2 -lvorbis -lvorbisenc -logg -lsamplerate \
 	-lx264 -lmp4v2 -lswscale -ltheora -lz \
 	-lbz2 -liberty -lpthreadGC2 -lbluray -lass -lfontconfig -lfreetype
 else
 HB_LIBS= \
 	-lhb -la52 -lmkv -lavformat -lavcodec -lavutil -ldca -ldvdnav -ldvdread \
-	-lfaac -lmp3lame -lmpeg2 -lvorbis -lvorbisenc -logg -lsamplerate \
+	-lmp3lame -lmpeg2 -lvorbis -lvorbisenc -logg -lsamplerate \
 	-lx264 -lmp4v2 -lswscale -ltheoraenc -ltheoradec -lz -lxml2 \
 	-lbz2 -lpthread -lbluray -lass -lfontconfig -lfreetype
 endif
--- a/gtk/src/audiohandler.c
+++ b/gtk/src/audiohandler.c
@@ -69,10 +69,7 @@
 		if (hb_audio_encoders[ii].encoder == fallback &&
 			!(hb_audio_encoders[ii].muxers & mux))
 		{
-			if ( mux == HB_MUX_MKV )
 				fallback = HB_ACODEC_LAME;
-			else
-				fallback = HB_ACODEC_FAAC;
 			break;
 		}
 	}
@@ -130,7 +127,7 @@
 			return HB_ACODEC_LAME;
 
 		case HB_ACODEC_AAC_PASS:
-			return HB_ACODEC_FAAC;
+			return HB_ACODEC_FFAAC;
 
 		case HB_ACODEC_AC3_PASS:
 			return HB_ACODEC_AC3;
@@ -144,7 +141,6 @@
 	{
 		mask = 	HB_ACODEC_LAME |
 				HB_ACODEC_FFAAC |
-				HB_ACODEC_FAAC |
 				HB_ACODEC_AC3;
 	}
 	if ( mux == HB_MUX_MKV )
--- a/gtk/src/hb-backend.c
+++ b/gtk/src/hb-backend.c
@@ -3729,7 +3729,7 @@
 gint
 ghb_get_default_acodec()
 {
-	return HB_ACODEC_FAAC;
+	return HB_ACODEC_LAME;
 }
 
 static void
@@ -4485,13 +4485,9 @@
 			{
 				codec = HB_ACODEC_AC3;
 			}
-			else if (mux == HB_MUX_MKV)
-			{
-				codec = HB_ACODEC_LAME;
-			}
 			else
 			{
-				codec = HB_ACODEC_FAAC;
+				codec = HB_ACODEC_LAME;
 			}
 			value = ghb_lookup_acodec_value(codec);
 			ghb_settings_take_value(asettings, "AudioEncoder", value);
@@ -4505,7 +4501,7 @@
 			if (codec == HB_ACODEC_VORBIS)
 			{
 				a_unsup = "Vorbis";
-				codec = HB_ACODEC_FAAC;
+				codec = HB_ACODEC_LAME;
 			}
 		}
 		if (a_unsup)
--- a/gtk/src/preset_xlat.c
+++ b/gtk/src/preset_xlat.c
@@ -260,7 +260,6 @@
 
 static value_map_t acodec_xlat[] =
 {
-	{"AAC (faac)", "faac"},
 	{"AC3 Passthru", "ac3"},
 	{"MP3 (lame)", "lame"},
 	{"Vorbis (vorbis)", "vorbis"},
--- a/gtk/src/presets.c
+++ b/gtk/src/presets.c
@@ -2004,9 +2004,6 @@
 static value_map_t acodec_xlat[] =
 {
 	{"AAC (ffmpeg)", "ffaac"},
-	{"AAC (faac)", "faac"},
-	{"AAC (CoreAudio)", "faac"},
-	{"HE-AAC (CoreAudio)", "faac"},
 	{"AC3 (ffmpeg)", "ffac3"},
 	{"AC3 (ffmpeg)", "ac3"},
 	{"AC3", "ac3"},			// Backwards compatibility with mac ui
--- a/libhb/common.c
+++ b/libhb/common.c
@@ -66,7 +66,7 @@
   { "AAC (CoreAudio)",    "ca_aac",     HB_ACODEC_CA_AAC,       HB_MUX_MP4|HB_MUX_MKV },
   { "HE-AAC (CoreAudio)", "ca_haac",    HB_ACODEC_CA_HAAC,      HB_MUX_MP4|HB_MUX_MKV },
 #endif
-  { "AAC (faac)",         "faac",       HB_ACODEC_FAAC,         HB_MUX_MP4|HB_MUX_MKV },
+  { "AAC (faac)",         "faac",       HB_ACODEC_FAAC,                             0 },
   { "AAC (ffmpeg)",       "ffaac",      HB_ACODEC_FFAAC,        HB_MUX_MP4|HB_MUX_MKV },
   { "AAC Passthru",       "copy:aac",   HB_ACODEC_AAC_PASS,     HB_MUX_MP4|HB_MUX_MKV },
   { "AC3 (ffmpeg)",       "ffac3",      HB_ACODEC_AC3,          HB_MUX_MP4|HB_MUX_MKV },
@@ -415,22 +415,6 @@
             }
             break;
 
-        case HB_ACODEC_FAAC:
-            *low = 32 * channels;
-            if (samplerate > 24000)
-            {
-                *high = 160 * channels;
-                if (*high > 768)
-                    *high = 768;
-            }
-            else
-            {
-                *high = 96 * channels;
-                if (*high > 480)
-                    *high = 480;
-            }
-            break;
-
         case HB_ACODEC_FFAAC:
             *low = 32 * channels;
             if (samplerate > 24000)
@@ -1560,7 +1544,7 @@
 
     /* Initalize some sensable defaults */
     audiocfg->in.track = audiocfg->out.track = 0;
-    audiocfg->out.codec = HB_ACODEC_FAAC;
+    audiocfg->out.codec = HB_ACODEC_LAME;
     audiocfg->out.bitrate = -1;
     audiocfg->out.quality = HB_INVALID_AUDIO_QUALITY;
     audiocfg->out.compression_level = -1;
--- a/libhb/common.h
+++ b/libhb/common.h
@@ -857,7 +857,6 @@
 extern hb_work_object_t hb_decavcodeca;
 extern hb_work_object_t hb_decavcodecv;
 extern hb_work_object_t hb_declpcm;
-extern hb_work_object_t hb_encfaac;
 extern hb_work_object_t hb_enclame;
 extern hb_work_object_t hb_encvorbis;
 extern hb_work_object_t hb_muxer;
--- a/libhb/encfaac.c
+++ b/libhb/encfaac.c
@@ -1,3 +1,4 @@
+#if 0
 /* $Id: encfaac.c,v 1.13 2005/03/03 17:21:57 titer Exp $
 
    This file is part of the HandBrake source code.
@@ -316,3 +317,4 @@
     return HB_WORK_OK;
 }
 
+#endif /* 0 */
--- a/libhb/hb.c
+++ b/libhb/hb.c
@@ -455,7 +455,6 @@
 	hb_register( &hb_decavcodeca );
 	hb_register( &hb_decavcodecv );
 	hb_register( &hb_declpcm );
-	hb_register( &hb_encfaac );
 	hb_register( &hb_enclame );
 	hb_register( &hb_encvorbis );
 	hb_register( &hb_muxer );
@@ -555,7 +554,6 @@
 	hb_register( &hb_decavcodeca );
 	hb_register( &hb_decavcodecv );
 	hb_register( &hb_declpcm );
-	hb_register( &hb_encfaac );
 	hb_register( &hb_enclame );
 	hb_register( &hb_encvorbis );
 	hb_register( &hb_muxer );
--- a/libhb/internal.h
+++ b/libhb/internal.h
@@ -360,7 +360,6 @@
     WORK_DECAVCODEC,
     WORK_DECAVCODECV,
     WORK_DECLPCM,
-    WORK_ENCFAAC,
     WORK_ENCLAME,
     WORK_ENCVORBIS,
     WORK_ENC_CA_AAC,
--- a/libhb/module.defs
+++ b/libhb/module.defs
@@ -1,4 +1,4 @@
-__deps__ := A52DEC BZIP2 FAAC FFMPEG FONTCONFIG FREETYPE LAME LIBASS LIBDCA \
+__deps__ := A52DEC BZIP2 FFMPEG FONTCONFIG FREETYPE LAME LIBASS LIBDCA \
     LIBDVDREAD LIBDVDNAV LIBICONV LIBMKV LIBOGG LIBSAMPLERATE LIBTHEORA LIBVORBIS LIBXML2 \
     MP4V2 MPEG2DEC PTHREADW32 X264 ZLIB LIBBLURAY
 
@@ -92,7 +92,7 @@
 LIBHB.lib = $(LIBHB.build/)hb.lib
 
 LIBHB.dll.libs = $(foreach n, \
-        a52 ass avcodec avformat avutil dca dvdnav dvdread faac fontconfig freetype mkv mpeg2 mp3lame mp4v2 \
+        a52 ass avcodec avformat avutil dca dvdnav dvdread fontconfig freetype mkv mpeg2 mp3lame mp4v2 \
         ogg samplerate swscale theora vorbis vorbisenc x264 xml2 bluray, \
         $(CONTRIB.build/)lib/lib$(n).a )
 
--- a/libhb/muxmkv.c
+++ b/libhb/muxmkv.c
@@ -255,7 +255,6 @@
                 }
                 track->codecID = MK_ACODEC_FLAC;
                 break;
-            case HB_ACODEC_FAAC:
             case HB_ACODEC_FFAAC:
             case HB_ACODEC_CA_AAC:
             case HB_ACODEC_CA_HAAC:
--- a/libhb/work.c
+++ b/libhb/work.c
@@ -128,7 +128,6 @@
     hb_work_object_t * w;
     switch( codec )
     {
-        case HB_ACODEC_FAAC:   return hb_get_work( WORK_ENCFAAC );
         case HB_ACODEC_LAME:   return hb_get_work( WORK_ENCLAME );
         case HB_ACODEC_VORBIS: return hb_get_work( WORK_ENCVORBIS );
         case HB_ACODEC_CA_AAC: return hb_get_work( WORK_ENC_CA_AAC );
--- a/test/module.defs
+++ b/test/module.defs
@@ -10,7 +10,7 @@
 TEST.exe = $(BUILD/)$(call TARGET.exe,$(HB.name)CLI)
 
 TEST.libs = $(LIBHB.a)
-TEST.libs += -lass -lfaac -lmp3lame -lsamplerate -lvorbisenc -ltheoraenc -ltheoradec -lvorbis -lx264 -lmkv -lz -lbz2 -lpthread -ldl -lmp4v2 -lxml2 -ldca -la52 -lbluray -lavcodec -lavformat -lavutil -lswscale -lmpeg2 -ldvdnav -ldvdread
+TEST.libs += -lass -lmp3lame -lsamplerate -lvorbisenc -ltheoraenc -ltheoradec -lvorbis -lx264 -lmkv -lz -lbz2 -lpthread -ldl -lmp4v2 -lxml2 -ldca -la52 -lbluray -lavcodec -lavformat -lavutil -lswscale -lmpeg2 -ldvdnav -ldvdread
 
 TEST.install.exe = $(DESTDIR)$(PREFIX/)bin/$(notdir $(TEST.exe))
 
